src/app/model/Queen.ts

```diff
@@ -0,0 +1,251 @@
+import { PiecePositions } from './PiecesPositions';
+import { BOARDSIZE } from './Constances';
+import { Piece } from './Piece';
+import { columns, Row, Side, Square } from './Types';
+import { PiecePossibleMoves } from './PiecePossibleMoves';
+
+export class Queen extends Piece implements PiecePossibleMoves {
+  constructor(side: Side) {
+    super(side);
+  }
+
+  possibleMoves(position: Square, board: PiecePositions): Square[] {
```

Comment:
Pierwsze koło ratunkowe. Wszystkie te metody np. widziałbym jako jedna o nazwie np. lineMoves (ze idzie po linii).

```
  possibleMoves(position: Square, board: PiecePositions): Square[] {
    return [
      ...lineMoves(board, position, {row: -1}),
      ...lineMoves(board, position, {row: 1}),
      ...lineMoves(board, position, {col: -1}),
      ...lineMoves(board, position, {col: 1}),
      ...lineMoves(board, position, {row: -1, col: -1}),
      ...lineMoves(board, position, {row: -1, col: 1}),
      ...lineMoves(board, position, {row: 1, col: 1}),
      ...lineMoves(board, position, {row: 1, col: -1})
    ]
  }
```

@tomdworniczak Czy to cos pomaga :) ?

Reply:
destrukturyzacja ładniej wygląda :P

Reply:
Ale pomogło?
Jakby sie uprzec to z tych 220 linijek metod moznaby zrobic ok. 5 :P 
Ale przyjmijmy jakies rozwiazanie najszybsze to tak nie wiecej niz 20 :)